lib/routing/sfRoute.class.php

```diff
@@ -279,6 +270,45 @@ public function generate($params, $context = [], $absolute = false)
         return $url;
     }
 
+    /**
+     * Generates a consistent cache key for a route
+     */
+    protected function generateCacheKey(array $elements = [], bool $absolute = false): string
+    {
+        $elements = [
+            SharedCacheHelper::ROUTING_NAMESPACE,
+            $this->flattenArrayElements($elements),
```

Comment:
This fixes the problem. By flattening (with recursion, if necessary) both the keys and values down into a single numerically-indexed array, the above examples no longer cause incorrect cache hits with one another.